one unbound config for all OS, fix #602

HenryNe opened the pull request:
Fix for #602
Create config `/etc/unbound/unbound.conf.d/openvpn.conf`,
include this config into `/etc/unbound/unbound.conf`,
do it for all OS.

On uninstall simple remove the include, if if was injected by openvpn-install.

_edit:_
tested on Debian 9 and Arch Linux 1.4

HenryNe committed:
one unbound config for all OS

Fix for #602
Create new config as /etc/unbound/unbound.conf.d/openvpn.conf,
include this into /etc/unbound/unbound.conf.
On uninstall simple remove the include, if if was injected by openvpn-install.

randshell commented on openvpn-install.sh:
```diff
@@ -1128,8 +1096,8 @@ function revokeClient () {
 	# Remove OpenVPN-related config
-	sed -i 's|include: \/etc\/unbound\/openvpn.conf||' /etc/unbound/unbound.conf
-	rm /etc/unbound/openvpn.conf
+	sed -i '/include .*\/openvpn.conf/d' /etc/unbound/unbound.conf
```
Can we use `/etc/unbound/unbound.conf.d/openvpn.conf` instead of wildcard?

HenryNe replied:
Yes. It's only longer, would not find the old location `/etc/unbound/openvpn.conf`, and needs lot of backslashes  ;-)
`
sed -i '/include \/etc\/unbound\/unbound.conf.d\/openvpn.conf/d' /etc/unbound/unbound.conf
`

randshell replied:
I think it's clearer to specify the full path but I'm fine anyway.

randshell commented on openvpn-install.sh:
```diff
@@ -146,48 +124,38 @@ prefetch: yes' >> /etc/unbound/unbound.conf
-	hide-version: yes
-	qname-minimisation: yes
-	prefetch: yes' > /etc/unbound/unbound.conf
+	qname-minimisation: yes' >/etc/unbound/unbound.conf
```
`harden-glue: yes` and `qname-minimisation: yes` are for all OS not only Arch. On `unbound.conf` manual they default to `yes` anyway.

randshell commented on openvpn-install.sh:
```diff
@@ -146,48 +124,38 @@ prefetch: yes' >> /etc/unbound/unbound.conf
+prefetch: yes' >/etc/unbound/unbound.conf.d/openvpn.conf
+
+	if [[ ! "$OS" =~ (fedora|centos|amzn) ]];then
+		# DNS Rebinding fix
```
DNS rebinding fix is for all OS not only fedora, centos or amzn

HenryNe replied:
OK. Have copied from old, and I would not change function here.

HenryNe committed:
harden-glue: and qname-minimisation for all OS

harden-glue: yes and qname-minimisation: yes for all OS, not only for Arch.

HenryNe committed:
DNS Rebinding fix for all OS

HenryNe committed:
Full path to remove include openvpn.conf

Full path to remove include of /etc/unbound/unbound.conf.d/openvpn.conf,
and missing ":" after the "include".

HenryNe committed:
Remove intermediate echo

randshell reviewed (changes requested):
Can you update the path to our config in the questions please?
https://github.com/angristan/openvpn-install/blob/e5f169ad3b94ddbb12cec24b5fa46135bd89eb9b/openvpn-install.sh#L270

HenryNe committed:
Text unbound.conf.d/openvpn.conf

randshell reviewed (approved):
@HenryNe Looks good.
Now I'm waiting for feedback from @angristan to resolve my previous comments.

HenryNe committed:
arch: username unbound default

"username: unbound" is default. Found in arch linu xdefauls unbound.conf and https://www.nlnetlabs.nl/documentation/unbound/unbound.conf/#username

HenryNe committed:
arch-Linux: Remove defaults

use-syslog is default, directory /etc/unbound is default, port 53 is default

HenryNe committed:
Arch: do-daemonize no is default

Arch Linux starts "/usr/bin/unbound -d -p".
"-d" do not fork into the background.
So the "daemonize: no" is default. (For Systemd on Arch Linux)

HenryNe committed:
openvpn.conf: TAB/ident

All entries after "server:" with a TAP/ident

HenryNe committed:
Merge branch 'master' into patch-7

HenryNe committed:
Merge changes from #591

HenryNe committed:
Merge branch 'master' into patch-7

HenryNe committed:
Merge branch 'master' into patch-7
